src/rustdoc-json-types/lib.rs

```diff
@@ -300,24 +303,68 @@ pub struct Struct {
 pub struct Enum {
     pub generics: Generics,
     pub variants_stripped: bool,
+    /// Will all be [`ItemEnum::Variant`]
     pub variants: Vec<Id>,
     pub impls: Vec<Id>,
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
-#[serde(rename_all = "snake_case")]
-#[serde(tag = "variant_kind", content = "variant_inner")]
-pub enum Variant {
-    Plain,
-    Tuple(Vec<Type>),
-    Struct(Vec<Id>),
+pub struct Variant {
+    pub kind: StructKind,
+    /// Will all be [`ItemEnum::Field`]
+    pub fields: Vec<Id>,
+    pub fields_stripped: bool,
 }
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
 #[serde(rename_all = "snake_case")]
-pub enum StructType {
-    Plain,
+/// The kind used for literals of a struct or enum.
+pub enum StructKind {
+    /// A "normal" struct, with named fields.
+    ///
+    /// Eg:
+    ///
+    /// ```rust
+    /// pub struct A {
+    ///     x: i32,
+    /// }
+    /// pub struct B {}
+    /// pub enum C {
+    ///     Variant { x: i32 },
+    /// }
+    /// pub enum D {
+    ///     Variant {},
+    /// }
+    /// ```
+    NamedFields,
+    /// Unnamed fields, accessed with a number.
+    ///
+    /// Eg:
+    ///
+    /// ```rust
+    /// pub struct A(i32);
+    /// pub struct B();
+    /// pub enum C {
+    ///     Variant(i32),
+    /// }
+    /// pub enum D {
+    ///     Variant(),
+    /// }
+    /// ```
     Tuple,
+    /// No fields, and no parentheses.
+    ///
+    /// Note that this doesn't include no fields but with parenthesis, as seen
```

Comment:
You made the "parentheses" typo here too. Also, could you please remove the double negation? I know it's something that is valid in english (as far as I know) but it's quite tricky for non-english readers.